api/environments/identities/traits/models.py

```diff
@@ -3,14 +3,27 @@
 from django.core.exceptions import ObjectDoesNotExist
 from django.db import models
 from django.utils.encoding import python_2_unicode_compatible
-from simple_history.models import HistoricalRecords
+from django_lifecycle import AFTER_DELETE, AFTER_SAVE, LifecycleModel, hook
 
 from environments.identities.traits.exceptions import TraitPersistenceError
 from environments.models import BOOLEAN, FLOAT, INTEGER, STRING
 
 
+class TriatQuerySet(models.QuerySet):
+    def delete(self, *args, **kwargs):
+        super(TriatQuerySet, self).delete(*args, **kwargs)
+        # update all the affected identity documents in dynamodb
+        self.first().identity.bulk_send_to_dynamodb([trait.identity for trait in self])
```

Comment:
This looks like a very slow operation to me. @dabeeeenster  @matthewelwell  might be worth looking into storing traits as  a separate document?